test/integration/migrate/index.js

```diff
@@ -268,12 +221,52 @@ module.exports = function(knex) {
 
   });
 
+  describe('knex.migrate.latest in parallel', function() {
+
+    afterEach(function() {
+      return knex.migrate.rollback({directory: 'test/integration/migrate/test'});
+    });
+
+    if (knex.client.dialect === 'postgres') {
+      it("is able to run two migrations in parallel in postgres", function () {
```

Comment:
@itaysabato As I said in earlier comment and is said in the mysql specs linked: In our case auto-commit is disabled because we are selecting for update inside transaction.

When other transaction sees that row is already locked and tries to select it for update the DB should just wait until lock is freed instead of throwing an errors (the same way that postgresql does).

Since I still don't understand/believe that this won't work in mysql, I need to read more about that and I'll run some tests locally to see how it works and if it really throws an error or wait if two transactions are trying to select for update the same row.

Also another thing is oracle support, which should be also doable... I would like to get this test enabled to every dialect except sqlite. I'll follow up later on when I have some data how stuff really works after playing in mysql shell.